sycl/test/esimd/esimd_verify.cpp

```diff
@@ -1,18 +1,24 @@
-// RUN: not %clangxx -fsycl -fsycl-device-only -S %s -o /dev/null 2>&1 | FileCheck %s
-// RUN: not %clangxx -fsycl -fsycl-device-only -O0 -S %s -o /dev/null 2>&1 | FileCheck %s
+// RUN: not %clangxx -fsycl -fno-sycl-esimd-force-stateless-mem -fsycl-device-only -S %s -o /dev/null 2>&1 | FileCheck -check-prefix=CHECK-NEGATIVE %s
+// RUN: not %clangxx -fsycl -fno-sycl-esimd-force-stateless-mem -fsycl-device-only -O0 -S %s -o /dev/null 2>&1 | FileCheck -check-prefix=CHECK-NEGATIVE %s
+
+// RUN: not %clangxx -fsycl -fsycl-esimd-force-stateless-mem -fsycl-device-only -O0 -S %s -o /dev/null  2>&1 | FileCheck -check-prefix=CHECK-POSITIVE %s
 
 #include <sycl/ext/intel/esimd.hpp>
 #include <sycl/sycl.hpp>
 
 using namespace sycl;
 using namespace sycl::ext::intel::esimd;
 
-// CHECK-DAG: error: function 'int* sycl::_V1::accessor<{{.+}}>::get_pointer<{{.+}}>() const' is not supported in ESIMD context
-// CHECK-DAG: error: function '{{.+}} sycl::_V1::accessor<{{.+}}>::operator[]<{{.+}}>({{.+}}) const' is not supported in ESIMD context
-// CHECK-DAG: error: function '{{.+}}combine(int const&)' is not supported in ESIMD context
+// CHECK-NEGATIVE-DAG: error: function 'int* sycl::_V1::accessor<{{.+}}>::get_pointer<{{.+}}>() const' is not supported in ESIMD context
+// CHECK-NEGATIVE-DAG: error: function '{{.+}} sycl::_V1::accessor<{{.+}}>::operator[]<{{.+}}>({{.+}}) const' is not supported in ESIMD context
+// CHECK-NEGATIVE-DAG: error: function '{{.+}}combine(int const&)' is not supported in ESIMD context
+
+// CHECK-POSITIVE-NOT: error: function 'int* sycl::_V1::accessor<{{.+}}>::get_pointer
```

Comment:
[CHECK-NOT](https://llvm.org/docs/CommandGuide/FileCheck.html#the-check-not-directive) is kind of confusing, it only checks that it does not find something between two other matches, i'm not sure this will error the way we want. maybe we need [implicit-check-not](https://llvm.org/docs/CommandGuide/FileCheck.html#cmdoption-filecheck-implicit-check-not)?

Reply:
Good advice. Thank you. Is this variant 34c3f9ec0de9a8afc71d79387ad5a0ac31c6a18e better?

Reply:
yeah i think the current version makes sure there is no `error: function` before or after the match for `error: function '{{.+}}combine(int const&)' is not supported in ESIMD context`